### PR TITLE
ENG-508: Create a Load and Soak testing framework

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **Status**: `draft` 
- **RFCs**: 
- **Categories**:  Utils Test

Develop mini framework for load and soak testing

Initial architecture definition:

```mermaid
flowchart TD
  subgraph Execute process
    PROCESS("process-compose:
     Entry point, 
     start all process")
  end
  subgraph Full Node
    SUZ(Suzuka Node)
    CEL(celestia)
    SEQ(Sequencer)
  end
  subgraph Tester
    SCH("Scheduler:
    - Schedule client execution.
    - Use rayon to schedule 
    - Each client in its own thread.
    - When a client execution end, return the result.
    - Store client execution info. Use formated logs" )
    CONF("Config:
    Define the test parameters")
    CLI("Client:
    - Execute several scenarios in async mode.
    - Log: scenario errors.
     ")
    SCE("Scenario:
    Define the scenario steps.")
  end

  PROCESS-- Start -->SUZ
  PROCESS-- Start -->CEL
  PROCESS-- Start -->SEQ
  PROCESS-- Start -->SCH
  CONF-- Define Exec -->SCH
  SCH-- Start Execution -->CLI
  CLI-- Run the scenario -->SCE
  SCE-- Return exec result -->CLI
  CLI-- Return exec result -->SCH
```
Scenario are developed in Rust and are linked with the Test framework. Scenario are defined in the bin folder of the crate.

The runtime start several clients that are executed using Rayon crate. Each client execute several scenarios using a mono thread Tokio async runtime.  For load test, the number of client is defined by the total number of scenario to start divided by the number of scenario per client, see ExecutionConfig. 

Execution data are written in a formatted (json) log file to be processed later if needed.
A function to log in this formatted file is provided in the scenario trait.

After the execution, the Test framework exe print on the stdout a final report.

I put the framework crate in `util/load-soak-testing`, see the changed files.

# Changelog

# Testing

# Outstanding issues

### CODE REVIEW

**Comment by musitdev** [comment]
I don't know why I get a lot of previous commit. I've started from main and rebase it. If you've an idea.
The developed files are in util/load_soak_testing

**Comment by l-monninger** [comment]
Overall this is good for now. But, please fix the typos and for organization, I would prefer to see the tests in the respective network or client crate instead of within the framework crate. I would prefer to include the framework where need and have the tests organized by service.

**Comment by musitdev** [comment]
I move the crate in networks/suzuka/load-soak-testting
I've implemented a simple soak test that loop and execute scenario during the test duration.
I've corrected some typos mistake.
The remaining part is to run new scenarios from time to time in a cycling way on top of the base looping ones.

File: networks/suzuka/load-soak-testing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+}
+
+impl ClientExecResult {
+	fn new(sceanarios: Vec<ScenarioExecMetric>) -> Self {
+		ClientExecResult {
+			average_execution_time_milli: Self::calcualte_average_exec_time_milli(&sceanarios),
+		}
+	}
+
+	pub fn calcualte_average_exec_time_milli(sceanarios: &[ScenarioExecMetric]) -> u128 {
```
Comment (l-monninger) [inline networks/suzuka/load-soak-testing/src/lib.rs:381]: There are still typos in here.

File: networks/suzuka/load-soak-testing/bin/basicalicebob.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
+use anyhow::{Context, Result};
+use aptos_sdk::{
+	coin_client::CoinClient,
```
Comment (l-monninger) [inline networks/suzuka/load-soak-testing/bin/basicalicebob.rs:3]: For organization, I still want you to move these out into the appropriate network crates. `suzuka-client` for example.

**Review by l-monninger — APPROVED** [review]

File: networks/suzuka/load-soak-testing/bin/demoscenario.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+use anyhow::Result;
```
Comment (l-monninger) [inline networks/suzuka/load-soak-testing/bin/demoscenario.rs:1]: Please fix typos and change organization before merging.

**Review by l-monninger — CHANGES_REQUESTED** [review]

**Comment by musitdev** [comment]
Recreate the PR in #111 to remove unknown modifications.

**Pull request closed without merging**